crates/net/network/src/transactions/mod.rs

```diff
@@ -940,19 +945,21 @@ where
                     Entry::Vacant(entry) => {
                         // this is a new transaction that should be imported into the pool
                         let pool_transaction = <Pool::Transaction as FromRecoveredPooledTransaction>::from_recovered_pooled_transaction(tx);
+                        to_add.push(pool_transaction);
 
-                        let pool = self.pool.clone();
-
-                        let import = Box::pin(async move {
-                            pool.add_external_transaction(pool_transaction).await
-                        });
-
-                        self.pool_imports.push(import);
                         entry.insert(vec![peer_id]);
```

Comment:
this makes me think that we can now get rid of this @emhane 

https://github.com/paradigmxyz/reth/blob/add3c6a3a2ae0cecd8627533bbc5688a456435ad/crates/net/network/src/transactions/mod.rs#L206-L210

Reply:
for the purpose of rate limiting tx requests to max inflight requests per tx hash at any given time, we can remove this. it's not needed anymore since `active_peers` and `inflight_requests` in `TransactionFecther` handle this. what we need to remove that data structure, is run the following code
https://github.com/paradigmxyz/reth/blob/add3c6a3a2ae0cecd8627533bbc5688a456435ad/crates/net/network/src/transactions/mod.rs#L941-L951
...somewhere here, where we have indented unknown previously unseen hashes (new hashes)
https://github.com/paradigmxyz/reth/blob/add3c6a3a2ae0cecd8627533bbc5688a456435ad/crates/net/network/src/transactions/mod.rs#L593

Reply:
well, the bellow batch code can be moved there to be precise

Reply:
https://github.com/paradigmxyz/reth/issues/6304